runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PCollectionTranslation.java

```diff
@@ -62,19 +62,19 @@ public static IsBounded isBounded(RunnerApi.PCollection pCollection) {
     return fromProto(pCollection.getIsBounded());
   }
 
-  static RunnerApi.IsBounded toProto(IsBounded bounded) {
+  static RunnerApi.IsBounded.Enum toProto(IsBounded bounded) {
     switch (bounded) {
       case BOUNDED:
-        return RunnerApi.IsBounded.BOUNDED;
+        return RunnerApi.IsBounded.Enum.BOUNDED;
       case UNBOUNDED:
-        return RunnerApi.IsBounded.UNBOUNDED;
+        return RunnerApi.IsBounded.Enum.UNBOUNDED;
       default:
         throw new IllegalArgumentException(
             String.format("Unknown %s %s", IsBounded.class.getSimpleName(), bounded));
     }
   }
 
-  static IsBounded fromProto(RunnerApi.IsBounded isBounded) {
+  static IsBounded fromProto(RunnerApi.IsBounded.Enum isBounded) {
```

Comment:
<!--new_thread; commit:6fddd4ed951c6e5618add72f49211bedcf046edc; resolved:1-->
I was really hoping for `IsBounded.IsBounded isBounded` and then maybe later we could even have assignments like `IsBounded.IsBounded isBounded = pcollection.isBounded()`

:-)

Reply:
Me too, but then found this pattern in several proto users code and it made a lot of sense.